src/Agent.Worker/Handlers/Handler.cs

```diff
@@ -68,13 +68,36 @@ protected void AddEndpointsToEnvironment()
                 AddEnvironmentVariable(
                     key: $"ENDPOINT_AUTH_{partialKey}",
                     // Note, JsonUtility.ToString will not null ref if the auth object is null.
-                    value: JsonUtility.ToString(endpoint.Authorization));
+                    value: JsonUtility.ToString(endpoint.Authorization)); 
+                if(endpoint.Authorization != null && endpoint.Authorization.Scheme != null)
+                {
+                    AddEnvironmentVariable(
+                        key: $"ENDPOINT_AUTH_SCHEME_{partialKey}",
+                        value: endpoint.Authorization.Scheme);
+
+                    foreach(KeyValuePair<string, string> pair in endpoint.Authorization.Parameters)
+                    {
+                        AddEnvironmentVariable(
+                            key: $"ENDPOINT_AUTH_PARAMETER_{partialKey}_{pair.Key?.Replace(' ', '_').ToUpperInvariant()}",
+                            value: pair.Value);
+                    }
```

Comment:
It doesn't make sense to set endpoint auth parameters without the scheme, so I put them together.